diffsims/generators/rotation_list_generators.py

```diff
@@ -159,83 +161,521 @@ def get_grid_around_beam_direction(beam_rotation, resolution, angular_range=(0,
     return rotation_list
 
 
-def get_beam_directions_grid(crystal_system, resolution, equal="angle"):
+def normalize_vectors(vectors):
     """
-    Produces an array of beam directions, evenly (see equal argument) spaced that lie within the streographic
-    triangle of the relevant crystal system.
+    Helper function which returns a list of vectors normalized to length 1 from
+    a 2D array representing a list of 3D vectors
+    """
+    return (vectors.T/np.linalg.norm(vectors, axis=1)).T
+
+
+def get_uv_sphere_mesh_vertices(resolution):
+    """
+    Return the vertices of a UV (polar) mesh on a unit sphere.
+    The mesh vertices are defined by the parametrization:
+
+    x = sin(u)cos(v)
+    y = sin(u)sin(v)
+    z = cos(u)
 
     Parameters
     ----------
-    crystal_system : str
-        Allowed are: 'cubic','hexagonal','trigonal','tetragonal','orthorhombic','monoclinic','triclinic'
-
     resolution : float
-        An angle in degrees. If the 'equal' option is set to 'angle' this is the misorientation between a
-        beam direction and its nearest neighbour(s). For 'equal'=='area' the density of points is as in
-        the equal angle case but each point covers an equal area
-
-    equal : str
-        'angle' (default) or 'area'
+        An angle in degrees. The maximum angle between nearest neighbor
+        grid points. In this mesh this occurs on the equator of the sphere.
+        All elevation grid lines are separated by at most resolution. The
+        step size of u and v are rounded up to get an integer number of
+        elevation and azimuthal grid lines with equal spacing.
 
     Returns
     -------
-    points_in_cartesians : np.array (N,3)
-        Rows are x,y,z where z is the 001 pole direction.
+    points_in_cartesian : np.array (N,3)
+        Rows are x, y, z where z is the 001 pole direction
+
+    References
+    ----------
+    https://medium.com/game-dev-daily/four-ways-to-create-a-mesh-for-a-sphere-d7956b825db4
     """
-    steps_theta = int(np.ceil(180 / resolution))  # elevation
-    steps_psi = int(np.ceil(360 / resolution))  # azimuthal
+    steps_theta = int(np.ceil(180 / resolution))+1  # elevation
```

Comment:
When I was testing for maximum nearest neighbor angle, I found that the elevation gridlines are actually slightly more widely spaced than `resolution`. I think it comes from the lines below these lines where the linspace is generated using or not using the end point. Adding just 1 extra gridline resolved the issue with the tests, but I didn't bother looking more closely into it.

Reply:
I went back just to check and indeed it's because on theta you state `endpoint=True`. So say you want a resolution of 10 degrees, you should have `steps_theta=19` not 18, because you include both endpoints: 0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120, 130, 140, 150, 160, 170, 180 = 19 values. So I'm leaving it with the +1

Reply:
sounds good to me, glad to have that cleared up.